homeassistant/components/light/insteon_hub.py

```diff
@@ -47,18 +55,29 @@ def brightness(self):
         """Return the brightness of this light between 0..255."""
         return self._value / 100 * 255
 
+    def insteon_update(self, _):
+        """Callback for an update from the hosue streaming endpoint"""
+        self.update_ha_state()
+
     def update(self):
         """Update state of the sensor."""
         resp = self.node.send_command('get_status', wait=True)
         try:
             self._value = resp['response']['level']
+            new_status = 'on' if self._value != 0 else 'off'
+            self.node.set_status(new_status)
```

Comment:
This is setting a cached status of the device in the package. It does not assume a status based on the response as responses are very broad and can be interpreted different ways.